cantabular-import/helpers/extract-job-info/extract-job-info.go

```diff
@@ -108,8 +111,43 @@ func main() {
 			return fieldsi[1] < fieldsj[1]
 		})
 
+		gotFirstTime := false
+		maxLastTime := ""
+		var maxDiff time.Duration
+		maxDiffTime := ""
+
 		for _, line := range linesFound {
 			fields := strings.Fields(line)
+
+			if gotFirstTime {
+				f, ferr := time.Parse(time.RFC3339, maxLastTime) // time format with nanoseconds
+				if ferr != nil {
+					fmt.Printf("error in maxLastTime: %v\n", ferr)
+				}
+
+				l, lerr := time.Parse(time.RFC3339, fields[1])
+				if lerr != nil {
+					fmt.Printf("error in fields[1]: %v\n", lerr)
+				}
+
+				if ferr == nil && lerr == nil {
+					fTime := f.Local()
+					lTime := l.Local()
+
+					diffNanoseconds := lTime.Sub(fTime)
+					diffsFound = append(diffsFound, diffNanoseconds)
+					if diffNanoseconds > maxDiff {
+						maxDiff = diffNanoseconds
+						maxDiffTime = fields[1]
+					}
+					maxLastTime = fields[1]
+					printAndSave(idResultFile, fmt.Sprintf("time since last id: %d.%09d seconds", diffNanoseconds/1000000000, diffNanoseconds%1000000000))
```

Comment:
I would suggest using the `Seconds()` function (implemented by the Duration struct) instead, which does the same:
```
// Seconds returns the duration as a floating point number of seconds.
func (d Duration) Seconds() float64 {
	sec := d / Second
	nsec := d % Second
	return float64(sec) + float64(nsec)/1e9
}
```

Reply:
Done